Feature/3

StanislavDargel opened the pull request.

StanislavDargel committed:
#feature/2 and feature/3
package refactoring:
1. ShoppingListApplication.java splitted into ConsoleUI.java and ShoppingListApplication.java. From ShoppingListApplication.java all application function logic moved into ConsoleUI.java.
1.1. Created package <console>, ConsoleUI.java refactored into <console> package.
2. Created package <domain>, Product.java and ProductCategory.java refactored into <domain> package.
3. ProductRepository.java refactored into ProductRepositoryImp.java. Created interface ProductRepository.java. ProductRepositoryImp.java implements ProductRepository.java.
3.1. Created package <repository>, ProductRepository.java and ProductRepositoryImp.java refactored into <repository>.
3.2. In ProductRepositoryImp.java inMemoryDatabase key changed from Long -> String. Methods fixed.
4. Created interface ProductValidationRule.java, ProductNotRemovedException.java, ProductNotFoundException.java. ProductValidationService.java methods splitted into ProductCategoryValidationRule.java, ProductDescriptionValidationRule.java, ProductDiscountValidationRule.java, ProductNameValidationRule.java and ProductPriceValidationRule.java.
4.1. Created package <validation>, ProductValidationService.java, ProductValidationRule.java, ProductCategoryValidationRule.java implements ProductValidationRule, ProductDescriptionValidationRule.java implements ProductValidationRule, ProductDiscountValidationRule.java implements ProductValidationRule, ProductNameValidationRule.java implements ProductValidationRule, ProductPriceValidationRule.java implements ProductValidationRule, ProductValidationExceptions.java, ProductNotRemovedException.java, ProductNotFoundException.java refactored into <validation>.
4.1.1. In ProductValidationService.java created Set collection, added productValidation(Product product) method.
4.2. Created package <service>, ValidationService.java and <validation> refactored into <service>.
4.2.1. In ValidationService.java methods fixed.

StanislavDargel committed:
#feature/2 and feature/3
package refactoring:
1. ShoppingListApplication.java splitted into ConsoleUI.java and ShoppingListApplication.java. From ShoppingListApplication.java all application function logic moved into ConsoleUI.java.
1.1. Created package <console>, ConsoleUI.java refactored into <console> package.
2. Created package <domain>, Product.java and ProductCategory.java refactored into <domain> package.
3. ProductRepository.java refactored into ProductRepositoryImp.java. Created interface ProductRepository.java. ProductRepositoryImp.java implements ProductRepository.java.
3.1. Created package <repository>, ProductRepository.java and ProductRepositoryImp.java refactored into <repository>.
3.2. In ProductRepositoryImp.java inMemoryDatabase key changed from Long -> String. Methods fixed.
4. Created interface ProductValidationRule.java, ProductNotRemovedException.java, ProductNotFoundException.java. ProductValidationService.java methods splitted into ProductCategoryValidationRule.java, ProductDescriptionValidationRule.java, ProductDiscountValidationRule.java, ProductNameValidationRule.java and ProductPriceValidationRule.java.
4.1. Created package <validation>, ProductValidationService.java, ProductValidationRule.java, ProductCategoryValidationRule.java implements ProductValidationRule, ProductDescriptionValidationRule.java implements ProductValidationRule, ProductDiscountValidationRule.java implements ProductValidationRule, ProductNameValidationRule.java implements ProductValidationRule, ProductPriceValidationRule.java implements ProductValidationRule, ProductValidationExceptions.java, ProductNotRemovedException.java, ProductNotFoundException.java refactored into <validation>.
4.1.1. In ProductValidationService.java created Set collection, added productValidation(Product product) method.
4.2. Created package <service>, ValidationService.java and <validation> refactored into <service>.
4.2.1. In ValidationService.java methods fixed.

javagururt commented on src/com/javaguru/shoppinglist/service/validation/ProductDiscountValidationRule.java:
```diff
@@ -0,0 +1,23 @@
+            } else if (product.getDiscount().compareTo(BigDecimal.ZERO) < 0 || product.getDiscount().compareTo(new BigDecimal(100.0)) > 0) {
+                throw new ProductValidationExceptions("Product discount is Incorrect, (Correct bounds (0 ~ 100.0))");
+            } else {
+                product.setDiscount(product.getDiscount().setScale(1, RoundingMode.HALF_EVEN));
```
метод валидации должен просто проверять. Ничего сетать метод "проверки" не должен

StanislavDargel committed:
#feature/3
1. In package <domain> in Product.java fixed line 92.
2. In package <service\validation> in ProductDiscountValidationRule.java, ProductNameValidationRule.java, ProductPriceValidationRule.java setters moved into package<console> in ConsoleUI.java.
3. In package <service\validation> created ProductActualPriceValidationRule.java.
4. ProductActualPriceValidationRule.java method addded to ProductValidationService.java.
5. In package <console> in ConsoleUI.java added setters to initialize correctly Product parameters. Fixed lines in createProduct() method, and in editProduvtParameters() method.
